src/main/java/org/opensrp/web/config/security/OAuth2SecurityConfig.java

```diff
@@ -89,12 +91,17 @@ public DefaultTokenServices tokenServices() {
 	public JdbcTokenStore tokenStore() {
 		final JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
 		final AuthenticationKeyGenerator authenticationKeyGenerator = new DefaultAuthenticationKeyGenerator();
+		Logger logger = LogManager.getLogger(JdbcTokenStore.class.toString());
 		return new JdbcTokenStore(dataSource) {
 			
 			@Override
 			public void storeAccessToken(final OAuth2AccessToken token, final OAuth2Authentication authentication) {
 				final String key = authenticationKeyGenerator.extractKey(authentication);
-				jdbcTemplate.update("delete from oauth_access_token where authentication_id = ?", key);
+				if( key == null || authentication == null)
```

Comment:
- I think the check for authentication needs to happen before passing the `authentication` object to the function
- Evaluate and see if super needs to be called even if we skip the delete operation

Reply:
super needs to be called since it is mostly operating on the `OAuth2AccessToken token`.